Barricades block leaps, lunges, melee attacks and tackles from the front

Arendian opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Adding barbed wire to a barricade now increases it's max health by 50.
Any xeno that attempts to leap or lunge over a barbed barricade will be stopped and stunned.
Trying to melee attack or tackle over a barricade from the front now makes you attack the barricade instead of the targeted entity. The chance for this to happen is the current health percentage of the barricade, with a minimum chance of 30%.

Tail Stab is now a targeted attack instead of a line.
Leap actions now ignore barricades without barbed wire and go over them.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
Parity.

## Technical details
<!-- Summary of code changes for easier review. -->
When an attack or tackle is attempted a raycast will check if there are any entities with the DirectionalAttackBlockerComponent in the way, if there are, a check is done to see if the attack is coming from the front of any found entities. The chance for the entity to block the attack is based on it's current health, every time it takes damage, or when the entity fails to block an attack, a new random number between 0 and 1 is generated. If the random number is higher than the current health percentage of the barricade, the next attack it receives will not be blocked.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

https://github.com/user-attachments/assets/b689e49c-104d-4015-aa88-eeff9973ea66


## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->
- [X] By submitting this code and/or assets, I confirm that I either own them or have provided the correct necessary licenses to use and distribute them. I agree to be fully responsible for any legal claims or issues arising from the use of these materials.

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl: Dygon
- tweak: Applying barbed wire to a barricade now increases it's max health by 50.
- tweak: Attempting to attack or tackle over a barricade from the front now has a chance to make you attack the barricade instead. The chance of this occurring ranges from 100% to 30%, depending on the barricade's current health.
- tweak: Trying to leap or pounce over a barbed barricade will now stun you.
- tweak: All leaps now ignore barricades without barbed wire.
- tweak: Tail Stab is now a targeted action and requires you to click on your target.
- fix: The grenadier armor's leap protection no longer blocks ravagers and praetorians.

Arendian commented:
> Does this allow runners and lurkers to leap over Cades if they're not barbed?

It currently does not, but I think changing the leap logic from adding an impulse to "throwing" the xeno will make them able to go over.

DrSmugleaf reviewed (changes requested):
You probably want to use deterministic rng like gun prediction/lag compensation does by using Xoroshiro64S instead, otherwise fast attacks will make the client and server roll different numbers and mispredict
If you do that you can put all the code in shared and not have to store a roll

Arendian commented:
I replaced the stored roll with a XoroShiro64 seed, logging shows the seed staying the same for the client and server. I also changed Tail Stab into a sprite click ability, it's now also blocked by barricades.

Arendian commented:
> Does this allow runners and lurkers to leap over Cades if they're not barbed?

Tables and barricades without barbed wire are now ignored during a leap.

DrSmugleaf reviewed (approved):
Thanks for the PR!
